src/node/types.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2010-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+//! @file node/types.h is a home for public enum and struct type definitions
+//! that are used by internally by node code, but also used externally by wallet
+//! or GUI code.
+//!
+//! This file is intended to define only simple types that do not have external
+//! dependencies. More complicated types should be defined in dedicated header
+//! files.
+
+#ifndef BITCOIN_NODE_TYPES_H
+#define BITCOIN_NODE_TYPES_H
+
+enum class TransactionError {
```

Comment:
It seems odd to put this in "node" when `TransactionErrorString(TransactionError error)` is in "common" ?

`TransactionError` is returned by `CombinePSBTs()` which seems like something that could reasonably be exposed by a command line tool without needing a full node setup.

So putting this in common seems more sensible to me?

Reply:
> With this background, I thought for this PR putting `TransactionError` in `node` instead of `common` would be better so it would be more likely to be appropriately used for functions like `node::BroadcastTransaction` and `interfaces::Node::broadTransaction`, and less likely to be misused for non-node functions. I could understand the opposite argument, though, that if even if it belongs in `node` when used ideally, it makes more sense for it to be in `common` based on the way it is used now.

I was mostly thinking that it doesn't make sense for `TransactionErrorString()` in common even though it depends on `TransactionError` which is in node, and node depends on common, not vice-versa? I wasn't expecting paragraphs of response :) I mean, it's just an enum with associated translatable descriptions? Doesn't really seem worth worrying about the fact that some of the errors only make sense in a node context; aren't these all just "something went wrong, deal with it, dear user"?

BTW, I think `TransactionError::P2P_DISABLED` can be dropped entirely? (I think the last/only use was removed in #15713)